UiPath.PowerShell.Tests/Cmdlets/MaintenanceModeTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+using UiPath.PowerShell.Models;
+using UiPath.PowerShell.Tests.Util;
+using UiPath.PowerShell.Util;
+using UiPath.Web.Client20198.Models;
+using Cmdlet = System.Management.Automation.PowerShell;
+
+namespace UiPath.PowerShell.Tests.Cmdlets
+{
+    /// <summary>
+    /// Maintenance Mode PowerShell tests
+    /// </summary>
+    /// <remarks> In order to run these, MaintenanceMode should be enabled in Orchestrator (add key="MaintenanceMode.Enabled" value="true")</remarks>
+    [TestClass]
+    public class MaintenanceModeTests : TestClassBase
+    {
+        private const string drainingPhase = "Draining";
+        private const string suspendedPhase = "Suspended";
+        private const string onlinePhase = "None";
+
+        [TestMethod]
+        public void MaintenanceMode_FullCycle_Works()
+        {
+            // Host admin op
+            using (var hostRunspace = PowershellFactory.CreateHostAuthenticatedSession(TestContext))
+            {
+                ICollection<PSObject> results;
+                try
+                {
+                    using (var cmdlet = PowershellFactory.CreateCmdlet(hostRunspace))
+                    {
+                        cmdlet.AddCommand(UiPathStrings.StartMaintentenanceMode)
+                            .AddParameter("Phase", drainingPhase)
+                            .AddParameter("Confirm", false);
+                        results = Invoke(cmdlet);
+                    }
+
+                    // check results in Draining
+                    var result = results.SingleOrDefault() as dynamic;
```

Comment:
nit, could just call Single() and remove null checks (if it fails it's a bug anyways)